engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportToSizeOf.java

```diff
@@ -65,12 +69,20 @@ public RenderPipelineTask generateTask() {
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(fboName);
+        // Generates a unique 32 bit signed integer from two integers.
+        // This will return unique values for the following reasonable ranges:
+        // width < (1 << 16) (65536) and height < (1 << 15) (32768)
+        return getWidth() << 15 | getHeight();
     }
 
     @Override
     public boolean equals(Object obj) {
-        return (obj instanceof SetViewportToSizeOf) && this.fboName.equals(((SetViewportToSizeOf) obj).getFboName());
+        if (!(obj instanceof SetViewportToSizeOf))
+            return false;
+
+        SetViewportToSizeOf other = (SetViewportToSizeOf) obj;
+
+        return getWidth() == other.getWidth() && getHeight() == other.getHeight();
```

Comment:
In fairness it does make the code in this particular case _a little_ easier to read. 

I'm tempted to say that we do not actually expect compared objects being anything else but StateChanges. So, potentially we could cast obj straight into a StateChange and skip the instanceof check. We'd only have to include in the javadoc of this method that it will generate an exception if obj isn't actually a StateChange. What do you guys think?

In any case state changes `SetBlendFunction`, `SetInputTexture `and `SetInputTextureFromFBO` could benefit from the same treatment.